system-server/system_server/system/register/router.py

```diff
@@ -0,0 +1,35 @@
+"""Router for /system/register endpoint."""
+from fastapi import APIRouter, Depends, status, Response
+from uuid import UUID
+
+from system_server.persistence import get_sql_engine, get_uuid
+from system_server.jwt import Registrant
+import sqlalchemy
+from .storage import get_or_create_registration_token
+from .models import PostRegisterResponse
+
+
+register_router = APIRouter()
+
+
+@register_router.post(
+    "/system/register",
+    summary="Register an agent with this robot.",
+    response_model=PostRegisterResponse,
+    status_code=status.HTTP_201_CREATED,
+)
+async def register_endpoint(
+    response: Response,
+    registrant: Registrant = Depends(Registrant),
+    signing_uuid: UUID = Depends(get_uuid),
```

Comment:
Just in case FastAPI tries to merge calls to `get_uuid()`:

```suggestion
    signing_uuid: UUID = Depends(get_uuid, use_cache=False),
```

https://fastapi.tiangolo.com/tutorial/dependencies/sub-dependencies/#using-the-same-dependency-multiple-times

Reply:
Caching would actually be fine here, though - this is referring to an ID that is persistent on the filesystem and the same value is used across the lifespan of this application.

This makes it obvious that the function `get_uuid` is too ambiguous though - it suggests that we're generating a new unique ID. Will rename.

Reply:
Ah, yeah, I definitely thought it was supposed to be unique. Thanks for the rename.